virtual_rainforest/core/model.py

```diff
@@ -135,10 +137,21 @@ def __init_subclass__(cls, model_name: str):
     def setup(self) -> None:
         """Function to use input data to set up the model."""
 
+    def start_model_timing(self, start_time: datetime64) -> None:
+        """Set of model timing by setting up initial last update time."""
+
+        if hasattr(self, "last_update"):
+            log_and_raise(
+                "Model timing was already set up, it should not be setup again!",
+                ImproperFunctionCall,
+            )
+        else:
+            self.last_update = start_time
+
     def should_update(self, current_time: datetime64) -> bool:
         """Determines whether a model should be updated for a specific time step."""
 
-        if current_time > self.last_update + self.update_interval:
+        if current_time >= self.last_update + self.update_interval:
             self.last_update = current_time
             return True
         return False
```

Comment:
I think rather than having a method which has to calculate for each time step, we can have `Model.update()` (I think I prefer that to `solve()` 😄 ) do stuff and then simply have: `self.next_update += self.update_interval`. See the sketch below for how this might work in the timing loop. That seems _much_ cheaper.